dataprofiler/data_readers/csv_data.py

```diff
@@ -107,6 +107,17 @@ def _check_and_return_options(self, options):
                                  'none for no header, or a non-negative '
                                  'integer for the row that represents the '
                                  'header (0 based index)')
+            if key == "delimiter" and value != None and not isinstance(value, str):
```

Comment:
Because it's starting to look like there will be more if statements than dictionary length, i think this should be changed from a forloop to several "if <key> in options.keys()" statements. Will look a little cleaner too.

Also, should use "is not None" instead of "!= None"